file/file.go

```diff
@@ -397,3 +397,73 @@ func HasPermission(fileName string, permission int) (bool, error) {
 
 	return true, nil
 }
+
+func CountLine(filename string) (uint64, error) {
+	return CountLineLogic("\n", filename, true)
+}
+
+func CountLineWithSeparator(separator, filename string) (uint64, error) {
+	return CountLineLogic(separator, filename, true)
+}
+
+// CountLineLogic function takes in a separator and a filename as a string, opens the file, and uses a bufio.Reader to read through the file.
+// It splits the file using the provided separator, and if skipEmptyLines is true, it skips empty lines.
+// It returns the number of lines in the file and any error that might have occurred.
+func CountLineLogic(separator, filename string, skipEmptyLines bool) (uint64, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return 0, err
+	}
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
+
+	var lineCount uint64
+	separatorRegexp := regexp.MustCompile(separator)
```

Comment:
Iterating over strings/bytes is way faster than using regexes, we should always prefer the native mechanism over compiled regexes (I'd suggest to move the regex handling into an independent function, where we also need to ensure that the regex can be actually compiled as MustCompile will trigger a panic otherwise)